### DISCUSSION TITLE
M5Stack Core 2 support

### DISCUSSION THREAD

**Discussion by gerard33** [HASwitchPlate/openHASP#67] [Hardware]
Will this software also work with the M5Stack Core 2 which is ESP32 based?
https://m5stack.com/products/m5stack-core2-esp32-iot-development-kit

The display is 2.0"@320*240 ILI9342C with a FT6336U touch screen.

I have one currently using https://github.com/sieren/Homepoint on the Core 2 but that’s not working as expected for the Core 2 at the moment.

**Comment by fvanroie** [comment]
There is basic support for the M5stack core2 in the source code. We have not released any binaries for this platform yet.
Some parts are not ready for the M5stack core2, but you can certainly try it out and provide your feedback.

To get your device flashed, you should install Visual Studio Code with PlarformIO and [build the configuration](https://fvanroie.github.io/hasp-docs/#compiling/) named `m5stack-core2`.

**Reply by gerard33** [reply]
Thx, will have a look at it and try this on my Core 2.

**Comment by Jaxom99** [comment]
I see M5Stack Core2 is now supported (worked well on my config), but I see there is a cheaper little brother : ESP32 Basic Core. It uses the same ESP32 processor, but replaces touchscreen for an ILI9342C with 3 physical buttons.

Do you think/know if OpenHASP could run on that, and manage both the screen and the buttons ? So far, I see that on the Core2 the "tactile buttons" are not handled....

**Reply by fvanroie** [reply]
I don't own a Basic, but I think it can work without touchscreen. The buttons are just GPIOs that can be read.

The Core2 doesn't support the virtual buttons because openHASP doesn't have a concept of virtual/ofscreen buttons...
